src/compiler.jl

```diff
@@ -1210,28 +1276,45 @@ end
 function runtime_pfor_rev(id::Int64)
     tape = leaked_objs[id]
     delete!(leaked_objs, id)
-	Base.Threads.threading_run(tape)
+    Base.Threads.threading_run(tape)
     return nothing
 end
-else 
-function runtime_pfor_augfwd(func, dfunc, dynamic)::Int64
-	# tape = vec{Any}(numthreads())
-	# pfor threads tape[i] = aug(func, dfunc)
-    
-	tt = Tuple{}
+else
+
+function runtime_pfor_fwd(func, dfunc, dynamic, width)::Cvoid
+    # pfor threads tape[i] = aug(func, dfunc)
+
+    tt = Tuple{}
     # TODO with the MI itself, we should be able to hoist the thunk generation into
     # the original compilation (rather than runtime JIT compiling)
-    forward, adjoint = thunk(func, dfunc, Const, tt, Val(API.DEM_ReverseModePrimal))
+    forward = thunk(func, dfunc, Const, tt, Val(API.DEM_ForwardMode), width)
 
-	tapes = Vector{Ptr{Cvoid}}(undef, Base.Threads.nthreads())
+    function fwd()
+        forward()
+    end
+
+    Base.Threads.threading_run(fwd, dynamic)
+    return
+end
+
+function runtime_pfor_augfwd(func, dfunc, dynamic, width)::Int64
+    # tape = vec{Any}(numthreads())
+    # pfor threads tape[i] = aug(func, dfunc)
+
+    tt = Tuple{}
+    # TODO with the MI itself, we should be able to hoist the thunk generation into
+    # the original compilation (rather than runtime JIT compiling)
+    forward, adjoint = thunk(func, dfunc, Const, tt, Val(API.DEM_ReverseModePrimal), width)
+
+    tapes = Vector{Ptr{Cvoid}}(undef, Base.Threads.nthreads())
     # tapes = unsafe_convert(Ptr{Ptr{Cvoid}}, Libc.malloc(sizeof(Ptr{Cvoid})*Base.Threads.nthreads))
-	function fwd()
-		tapes[Base.Threads.threadid()] = forward()[1]
-	end
-	adj = () -> adjoint(tapes[Base.Threads.threadid()])
+    function fwd(tid)
+        tapes[tid] = forward()[1]
+    end
+    adj = (tid) -> adjoint(tapes[tid])
```

Comment:
Are these threading fixes for 1.8? Split them out into a separate PR we will need to backport them